nf_core/sync.py

```diff
@@ -305,18 +280,25 @@ def update_existing_pull_request(self, pr_title, pr_body_text):
             r_json = r.content
             r_pp = r.content
 
-        # PR worked
         if r.status_code == 200:
             log.debug("GitHub API listing existing PRs:\n{}".format(r_pp))
 
             # No open PRs
             if len(r_json) == 0:
-                log.info("No open PRs found between TEMPLATE and {}".format(self.from_branch))
+                log.info("No open PRs found between {} and {}".format(branch, self.from_branch))
                 return False
 
-            # Update existing PR
+            # Close existing PR
+            pr_title = "Important! Template update for nf-core/tools v{} Closed because outdated!".format(
+                nf_core.__version__
+            )
+            pr_body_text = (
+                "A new release of the main template in nf-core/tools has just been released. "
+                "This automated pull-request attempts to apply the relevant updates to this pipeline.\n\n"
+                "This pull-request is outdated and has been closed. A new pull-request has been created instead."
+            )
             pr_update_api_url = r_json[0]["url"]
-            pr_content = {"title": pr_title, "body": pr_body_text}
+            pr_content = {"state": "closed", "title": pr_title, "body": pr_body_text}
 
             r = requests.patch(
```

Comment:
I don't think that this function checks for `os.environ.get("GITHUB_AUTH_TOKEN")`, right? So this will probably fail if running locally to do a manual sync.

I wonder if we should check for this once in the main `sync()` function and move this in with the other GitHub related functions. I guess it doesn't need to be done first, so we could also move it after the creation of the new PR in which case we can also link to the replacement which I think would be nice..

Reply:
Yes good points, checking for the token in `sync()` would be cleanest and makes sense.

And I'll try to switch the order and try to link to the replacement!